qiskit/algorithms/amplitude_amplifiers/amplification_problem.py

```diff
@@ -167,18 +166,21 @@ def is_good_state(self) -> Callable[[str], bool]:
             return self._is_good_state  # returns None if no is_good_state arg has been set
         elif isinstance(self._is_good_state, list):
             if all(isinstance(good_bitstr, str) for good_bitstr in self._is_good_state):
-                return lambda bitstr: bitstr in self._is_good_state
+                return lambda bitstr: bitstr in self._is_good_state  # type:ignore[operator]
```

Comment:
This shouldn't be ignored. Mypy should already know that `self._is_good_state` is an instance of `list`, and so is already doing type narrowing.

Reply:
This seems to be instance of https://github.com/python/mypy/issues/4297

Reply:
I think a safe path moving forward is to retain the Mypy error (not ignore it!), to be fixed in a future PR, and to add a TODO comment that links to https://github.com/python/mypy/issues/4297.